src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisShardSyncer.java

```diff
@@ -159,10 +164,15 @@ private synchronized void syncShardLeases(IKinesisProxy kinesisProxy,
      */
     // CHECKSTYLE:OFF CyclomaticComplexity
     private synchronized void syncShardLeases(IKinesisProxy kinesisProxy,
-            ILeaseManager<KinesisClientLease> leaseManager, InitialPositionInStreamExtended initialPosition,
-            boolean cleanupLeasesOfCompletedShards, boolean ignoreUnexpectedChildShards, List<Shard> latestShards)
+                                              ILeaseManager<KinesisClientLease> leaseManager,
+                                              InitialPositionInStreamExtended initialPosition,
+                                              boolean cleanupLeasesOfCompletedShards,
+                                              boolean ignoreUnexpectedChildShards,
+                                              List<Shard> latestShards,
+                                              Boolean isLeaseTableEmpty)
```

Comment:
We can just use the primitive `boolean` here

Reply:
The reason behind using the boxed version was to allow for `null`, which in this case means "I have no opinion on whether or not the lease table is empty, let the LeaseManager decide", whereas `false` means "I claim the lease table is not empty". Using primitives removes this option.

Let me know if you have any other ideas on how to approach this.

Reply:
Discussed offline - removed the boxed boolean and switched to primitive